doc/package-specific-notes.chapter.md

```diff
@@ -0,0 +1,52 @@
+---
+title: Package-specific notes
+author: Michael Raskin
+date: 2019-05-01
+---
+
+# Package-specific notes
+
+This chapters includes some notes that apply to specific packages and should
+answer some of the frequently asked questions.
+
+## OpenGL
+
+Packages that use OpenGL have NixOS desktop as their primary target. The
+current solution for loading the GPU-specific drivers is based on `libglvnd`
+and looks for the driver implementation in `LD_LIBRARY_PATH`. If you are using
+a non-NixOS GNU/Linux/X11 desktop with free software video drivers, consider
+launching OpenGL-dependent programs from Nixpkgs with Nixpkgs versions of
+`libglvnd` and `mesa_drivers` in `LD_LIBRARY_PATH`. For proprietary video
+drivers you might have luck with also adding the corresponding video driver
+package.
+
+## Locales
+
+To allow simultaneous use of packages linked against different versions of
+`glibc` with different locale archive formats Nixpkgs patches `glibc` to rely
+on `LOCALE_ARCHIVE` environment variable.
+
+On non-NixOS distributions this variable is obviously not set. This can cause
+regressions in language support or even crashes in some Nixpkgs-provided
+programs. The simplest way to mitigate this problem is exporting the
+`LOCALE_ARCHIVE` variable pointing to
+`${glibcLocales}/lib/locale/locale-archive`. The drawback (and the reason this
+is not the default) is the relatively large (a hundred MiB) size of the full
+set of locales. It is possible to build a custom set of locales by overriding
+parameters `allLocales` and `locales` of the package.
+
+## Unfree software
```

Comment:
This is going to make a lot of new users, who struggle to figure this out, lives a lot easier :+1:

Reply:
Initially I tried to also explain the reasons for the default state (not even visible to search), but decided that I don't want to give the remarks on unfree software too much prominence…

Reply:
I think that's a good call. The documentation itself should remain neutral and just serve to inform.

Reply:
Hmm there's also some stuff about `allowUnfree` at https://nixos.org/nixpkgs/manual/#chap-packageconfig

Reply:
I don't think rationale should never be explained (there are technical reasons), but I had various non-neutral goals about appearance of that paragraph (I wanted it to be slightly negative and not too prominent — to reflect the attitudes evident in the relevant discussions)

NixOS manual describes the setting in context of NixOS configuration (and it only affects NixOS evaluation there…), Nixpkgs manual should cover the setting in the context of Nixpkgs configuration for use via other means.

Reply:
Oh, it wasn't explicitly linked to #17126